IndexStatsIT#testFilterCacheStats fails

I think this failure relates to Lucene-7.5.

I can't reproduce this locally with this command:

```
./gradlew :server:integTest -Dtests.seed=528FC320ADC916D8 \
-Dtests.class=org.elasticsearch.indices.stats.IndexStatsIT \
-Dtests.method="testFilterCacheStats" \
-Dtests.security.manager=true \
-Dtests.locale=kk-KZ \
-Dtests.timezone=America/Scoresbysund
```

```
FAILURE 0.63s J0 | IndexStatsIT.testFilterCacheStats <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: a value greater than <0L>
   >      but: <0L> was equal to <0L>
   > 	at __randomizedtesting.SeedInfo.seed([528FC320ADC916D8:37B03E651DC111A1]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.indices.stats.IndexStatsIT.testFilterCacheStats(IndexStatsIT.java:1046)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   > 	at java.base/java.lang.Thread.run(Thread.java:834)
```

CI: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java11,nodes=virtual&&linux/210/consoleText

Log: [testFilterCacheStats.log](https://github.com/elastic/elasticsearch/files/2246264/testFilterCacheStats.log)

--- a/server/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java
+++ b/server/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java
@@ -19,10 +19,12 @@
 
 package org.elasticsearch.indices.stats;
 
+import com.carrotsearch.randomizedtesting.annotations.Repeat;
 import org.apache.lucene.util.LuceneTestCase.SuppressCodecs;
 import org.elasticsearch.action.DocWriteResponse;
 import org.elasticsearch.action.admin.cluster.node.stats.NodesStatsResponse;
 import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
+import org.elasticsearch.action.admin.indices.forcemerge.ForceMergeResponse;
 import org.elasticsearch.action.admin.indices.stats.CommonStats;
 import org.elasticsearch.action.admin.indices.stats.CommonStatsFlags;
 import org.elasticsearch.action.admin.indices.stats.CommonStatsFlags.Flag;
@@ -61,6 +63,7 @@ import org.elasticsearch.test.ESIntegTestCase;
 import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
 import org.elasticsearch.test.ESIntegTestCase.Scope;
 import org.elasticsearch.test.InternalSettingsPlugin;
+import org.elasticsearch.test.junit.annotations.TestLogging;
 
 import java.io.IOException;
 import java.util.ArrayList;
@@ -1006,7 +1009,8 @@ public class IndexStatsIT extends ESIntegTestCase {
         assertEquals(total, shardTotal);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/32506")
+    @Repeat(iterations = 1000)
+    @TestLogging("_root:TRACE")
     public void testFilterCacheStats() throws Exception {
         Settings settings = Settings.builder().put(indexSettings()).put("number_of_replicas", 0).build();
         assertAcked(prepareCreate("index").setSettings(settings).get());
@@ -1056,7 +1060,11 @@ public class IndexStatsIT extends ESIntegTestCase {
             persistGlobalCheckpoint("index");
             flush("index");
         }
+        ForceMergeResponse forceMergeResponse =
+            client().admin().indices().prepareForceMerge("index").setFlush(true).setMaxNumSegments(1).get();
+        assertAllSuccessful(forceMergeResponse);
         refresh();
+
         response = client().admin().indices().prepareStats("index").setQueryCache(true).get();
         assertCumulativeQueryCacheStats(response);
         assertThat(response.getTotal().queryCache.getHitCount(), greaterThan(0L));

--- a/server/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java
+++ b/server/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java
@@ -19,7 +19,6 @@
 
 package org.elasticsearch.indices.stats;
 
-import com.carrotsearch.randomizedtesting.annotations.Repeat;
 import org.apache.lucene.util.LuceneTestCase.SuppressCodecs;
 import org.elasticsearch.action.DocWriteResponse;
 import org.elasticsearch.action.admin.cluster.node.stats.NodesStatsResponse;
@@ -63,7 +62,6 @@ import org.elasticsearch.test.ESIntegTestCase;
 import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
 import org.elasticsearch.test.ESIntegTestCase.Scope;
 import org.elasticsearch.test.InternalSettingsPlugin;
-import org.elasticsearch.test.junit.annotations.TestLogging;
 
 import java.io.IOException;
 import java.util.ArrayList;
@@ -1009,8 +1007,6 @@ public class IndexStatsIT extends ESIntegTestCase {
         assertEquals(total, shardTotal);
     }
 
-    @Repeat(iterations = 1000)
-    @TestLogging("_root:TRACE")
     public void testFilterCacheStats() throws Exception {
         Settings settings = Settings.builder().put(indexSettings()).put("number_of_replicas", 0).build();
         assertAcked(prepareCreate("index").setSettings(settings).get());

